### ISSUE TITLE
Add more explanation to public view related code

### ISSUE THREAD

**Issue by Woseseltops** [Signbank/Global-signbank#1451]
When reviewing #1419 , I noticed there was a lot of code related to searching and 'dynamic languagues', which left me confused. As an example of something that confused me:

```python
        if not self.search_form.is_bound:
            # make sure the form has the dynamic language fields that require the request
            set_up_language_fields(LemmaIdgloss, self, self.search_form)
```

Why is the search form relevant to the public view? What does it mean for a search from 'to be bound'? What are dynamic language fields? What do they need to be set up, but only when the search from is bound?

Perhaps we can use this issue to
1. Better understand what is going on here
2. Create 1-2 sentence exlanation of this
3. Add this as a comment to the relevant parts in the code

**Comment by susanodd** [comment]
It's relevant to all views.

It's in that pull request because that's where it was encountered. (Otherwise I need to yank out the code and make yet another pull request with just that code.)

If the form hasn't been initialized with the languages yet, the search does not work yet. (The language fields are not in the form. They are put in the form once they are in the request from the selected dataset languages. Everything used to end up defaulting to NGT and it was not possible to browse in Kata Kolok public glosses because the languages had been set to Dutch and English, as for NGT.)

it's a result of cascading bugs related to the selected datasets getting set to NGT. For public view, the dataset selected is in the session variable. But that was not being used in the original code. (I was working on that together with @vanlummelhuizen last week.)

This has something to do with the order of evaluation of Django when it loads a view. The `get_queryset` is done before the `get_context_data` is done. The language fields need the "request" argument, but it is not available in the `__init__` method

The normal list views have a big "form your query" area at the top. Initially no results are shown.
That has the effect that by the time you type in your query the form has been initialised.

This has something to do with the order of evaluation of Django when it loads a view. The `get_queryset` is done before the `get_context_data` is done.

**Comment by susanodd** [comment]
I will add comments!!

**Comment by susanodd** [comment]
Can you guys fight it out?

I was fine with the original ONE-LINE comment! (See issue description.)

The Django `get_queryset` is called BEFORE then `get_context_data` method. I merely moved the initialization there.

Just yesterday, I had to move the initialization to the front of two views because it wasn't getting i initialized yet.
This depends on control flow. You have to actually be using Signbank, not just reading code to see why this is necessary.

**Comment by susanodd** [comment]
@Woseseltops complained about the ONE LINE COMMENT.

I changed it to be SIX LINES to be pedantic. Because he complained about the ONE LINE.

I'm going to change it back to what it was, just ONE line. I am fine with the original ONE LINE.

**Comment by vanlummelhuizen** [comment]
> Can you guys fight it out?

Sure. I talk to @Woseseltops on Thursday.

> I'm going to change it back to what it was, just ONE line. I am fine with the original ONE LINE.

Please wait to change things (back) until we settle this.

**Comment by Woseseltops** [comment]
> In short: if a form has data, it is bound (to that data), else it is unbound and may be used to create a blank form.

Ah didn't know, thanks :)

> Or change the function name to add_language_dependent_fields.

Even better!

> I changed it to be SIX LINES to be pedantic. Because he complained about the ONE LINE.

I detect frustration here, sorry the Signbank project is giving you negative emotions! I appreciate your attempt to improve the understandability of the code!

Having said that, if you read my original request, I'm not asking for more comments to the code immediately, I asked to start with formulating a short explanation together, here in this issue :). Short, for similar reasons I ask you to use the `<details>` tag.

Would the proposal by @vanlummelhuizen work for you?

**Comment by vanlummelhuizen** [comment]
> Would the proposal by @vanlummelhuizen work for you?

Yes. My suggestion would be to let the short explanation we come up with be reflected in the function name and function docstring. Then, if the function name is still unclear to the reader, he/she can hover it in the IDE to see the docstring. This way we wouldn´t need comments at all.

Rule of thumb: code should be self explanatory, by using good names and docstrings, and with the help of a good IDE.
